File: 451-sort-characters-by-frequency/sort-characters-by-frequency.cpp
class Solution {
public:
    static bool cmp(pair<char,int>& a, pair<char,int>& b) {
        return a.second > b.second; 
    }
    string frequencySort(string s) {
        int n = s.size();
        map<char,int>mpp;
        for(int i = 0 ; i < n ; i++){
            mpp[s[i]]++;
        }
        vector<pair<char,int>>v(mpp.begin(),mpp.end());
        sort(v.begin(),v.end(),cmp);
        string ans="";
        for(auto it:v){
            int k = it.second;
            while(k--){
                ans+=it.first;
            }
        }
        return ans;
    }
};

File: 451-sort-characters-by-frequency/sort-characters-by-frequency.py
class Solution:
    def frequencySort(self, s: str) -> str:
        l=[]
        c=Counter(s)
        for k,v in c.items():
            l.append(k*v)
        l.sort(key=len)
        l=l[::-1]
        return "".join(l)
    
    